### PR TITLE
Add image export sidebar panel

### PR DESCRIPTION
**Pull request by lossyrob**
mspc-landsat-c2-l2-63be31cf8bc54304b7c5ddcd3b52cbc2

This PR adds an image export option that builds on the work for animation export done in #343.

It refactors the sidebar state to show/hide components based on an Enum, which can now be SidebarPanels.itemSearch, SidebarPanels.animation, or SidebarPanels.image

It also refactors the `drawBbox` state on the mapSlice to be a `drawnShape`, which captures both the bbox and geometry from the atlas.Shape drawn on the map, as image export API uses a geometry parameter rather than a bbox. 

Also, I added social media share icons to the image viewer with the [next-share](https://github.com/Bunlong/next-share) library:
image.

This has LinkedIn, Twitter, and Facebook for now, but more could be added. We should check on guidelines for including shares to external providers. If we want to keep these, I can add them to the animation export viewer as well.

Note that most of the components are copied over from the AnimationExport, and there's likely refactoring to be done to consolidate some duplication. Looking for guidance on layout; feel free to add suggested refactors (e.g. combine this with the animation version and put at path X in the codebase) and I can clean things up.

Testing these changes through CI test deployments will require https://github.com/microsoft/planetary-computer-apis/pull/122 in PCT.

### CODE REVIEW

File: src/pages/Explore/components/Sidebar/ImageExporter/ImageViewer.tsx [hunk 1]
```diff
@@ -0,0 +1,153 @@
...
+            <TwitterIcon size={20} round />
+          </TwitterShareButton>
+          <FacebookShareButton
+            url={shareUrl}
+            quote={"next-share is a social share buttons for your next React apps."}
+            hashtag={"#nextshare"}
+          >
+            <FacebookIcon size={20} round />
+          </FacebookShareButton>
+        </Stack>
```
Comment (lossyrob) [inline src/pages/Explore/components/Sidebar/ImageExporter/ImageViewer.tsx:95]: TODO: Decide on best title/hashtags/quote text for social media shares.

File: src/pages/Explore/components/Sidebar/ImageExporter/ImageError.tsx [hunk 2]
```diff
@@ -0,0 +1,27 @@
...
+  error: AxiosError<any, any>;
+}
+
+export const ImageError: React.FC<ImageErrorProps> = ({ error }) => {
+  if (!error) return null;
+
+  const { response } = error;
+  const errorText =
+    response?.status === 400
+      ? response?.data.error
```
Comment (mmcfarland) [inline src/pages/Explore/components/Sidebar/ImageExporter/ImageError.tsx:14]: This doesn't look like the (current deployed) shape of a 400, I'm getting a JSON array of objects containing a `msg` property. Cross referenced the API PR and it doesn't look to get changed to .error there.

File: src/pages/Explore/components/Sidebar/ImageExporter/ImageExporter.index.tsx [hunk 3]
```diff
@@ -0,0 +1,322 @@
...
+    s => s.map
+  );
+  const images = useExploreSelector(s =>
+    selectImagesByCollection(s, collection?.id)
+  );
+  let imageSettings = useExploreSelector(s =>
+    selectImageSettings(s, collection?.id)
+  );
+
+  console.log(imageSettings);
```
Comment (mmcfarland) [inline src/pages/Explore/components/Sidebar/ImageExporter/ImageExporter.index.tsx:58]: A few straggler debug statements to remove.

**Review by mmcfarland — COMMENTED** [review]
This looks good!

I agree that there's probably a lot that could be consolidated, with some diminishing returns at a certain point. My recommendation for a first-pass would be to consolidate the `ImageResult`, `ImageResults`, and `ImageViewer` components into reusable set of components (besides using a single interface for the response type, they might not need any code changes, or at least very few). Consider a new directory structure like `Sidebar/exporters` that contains `BaseExporter`, `ImageExporter` etc. Then each concrete component would just import from the base folder -- for those three I'd even keep the same name, "ImageViewer" is generic enough to apply to both.

If you want to go one step further, I'd also pop the `*Intro.tsx` component to a `ExporterHeader` component that handled the top-level layout and the close logic, like:

```typescript
interface ExporterHeaderProps {
    title: string;
    subTitle: string;
    onClose: () => void;
}

...

<ExporterHeader
   title={"Export Image"} 
   subTitle={`${collection.title || collection.id}, ${renderOption?.name}`}
   onClose={handleClose}>
      <>
      <Text>
        The image will be generated using the current search. Draw a bounding box for
        the image and select the image size. The image returned will be at the size
        provided and will contain the area drawn in the bounding box. Note that the
        image bounds may be larger than the drawn area.
      </Text>
      <Text>Click on the generated image to download or share.</Text>
     </>
</ExporterHeader>
```

I don't think there's much value in trying to combine the UI/settings/state portion further.

**Pull request merged**